cryostatio/cryostat-grafana-dashboard
# feat(datasource): migrate simple-json to infinity datasource, upgrade Angular panels

Fixes #25 
Fixes #60

Current bugs:
~~1. The main dashboard panels are still placed into rows, but the rows are expanded by default and any chart that is in the viewport when the dashboard loads will try to render, triggering a datasource query, which can be expensive. Updating the dashboard JSON definition to collapse the rows leads to empty collapsed rows and visible panels rendered outside of the rows, but when this happens, expanding the empty row and collapsing it again causes it to "recapture" the chart panels and hide them.~~

On load:
![image](https://github.com/cryostatio/cryostat-grafana-dashboard/assets/3787464/30bcba65-1c41-4f89-be75-4cb6e1ab9cd0)

"Expand" Network row:
![image](https://github.com/cryostatio/cryostat-grafana-dashboard/assets/3787464/79ea8897-a459-45cb-a76e-5da74f79f940)

Collapse Network row:
![image](https://github.com/cryostatio/cryostat-grafana-dashboard/assets/3787464/d7522488-27b4-4328-9cf3-c1f6e9b3fb80)


2. ~~The object allocation [sub-dashboard's query for determining what classes have been allocated](https://github.com/cryostatio/cryostat-grafana-dashboard/pull/68/files#diff-d35f77b7acf4e880810a3b95d6e507092094c902bbf65af1bb83db4ca60acc3aR181) (setting the `objectClass` template variable) querys from the Unix epoch `0` (ie 1970-01-01-00:00:00) until 1000 years later. The `${__timeFrom:date}` and `${__timeTo:date}` macros that work in other datasource queries do not seem to work in the template variable query - the macros both always get substituted with the Unix epoch `0`, leading to a query with a range of 0 duration, and no samples, therefore an empty dashboard. The 1970-2970 range workaround works fine and will work for most of this millenium, and captures all of the allocated class names, so this is not actually causing any problem.~~

## To Test

1. Check out PR
2. `./build.sh && podman image prune -f`
3. `cd cryostat ; ./smoketest.bash -Ot` and wait for it to come up
4. Select `cryostat3:9091` target and "View in Grafana" on one of the existing recordings
5. The Grafana dashboard should appear. There should be no more Angular deprecation warnings. Otherwise, the dashboard should just look a feel a little "fresher" and maybe smoother, but function the same as before.

Before:
![image](https://github.com/cryostatio/cryostat-grafana-dashboard/assets/3787464/42e2bd52-dbe3-4f6c-8483-73ed3bfdcead)


After:
![image](https://github.com/cryostatio/cryostat-grafana-dashboard/assets/3787464/a3814b93-9c76-4c31-9da1-b0755d4a0e36)
